frontend/ui/widget/imagewidget.lua

```diff
@@ -443,6 +443,15 @@ function ImageWidget:getScaleFactorExtrema()
     return self._min_scale_factor, self._max_scale_factor
 end
 
+-- Compute a new best-fit scale factor for a given image rescale percentage
+function ImageWidget:recomputeScaleFactor(scaling)
+    local req_width = self._bb:getWidth() * scaling
+    local req_height = self._bb:getHeight() * scaling
+
+    -- Best fit
+    return math.min(req_width / self._img_w, req_height / self._img_h)
```

Comment:
Isn't self._bb:getWidth() == self._img_w (idem for height) ? Aren't A and B the same in the the math.min(A, B) ?

Reply:
`self._bb` is the *scaled* bb, while `self_img_*` is the unscaled image (i.e., the actual reference to compute the scale factor).

Reply:
i.e., the two only match in an hypopthetical scale_factor == 1 scenario.

Reply:
> Aren't A and B the same in the the math.min(A, B) ?

They ought to be since we're only concerned with best-fit here, yeah.

Reply:
OK, but we do not crop (and I'd like this to stay as is : zoom takes time, but panning is fast as it works with the same self._bb).
So, self._bb and the unscaled image keep the same A/R.
So, it looks to me that `req_width / self._img_w == req_height / self._img_h == scaling`.
(But I may miss something :) I'm just having a look at your diffs...)

Reply:
> So, it looks to me that `req_width / self._img_w == req_height / self._img_h == scaling`.

Yup, this is just duplicated because I felt like taking only a single dimension would feel arbitrary, and to match how it's actually computed the first time.

And to potentially handle the unlikely case where this ends up being called on a shady input ;).

Reply:
Well, it looks to me that it comes down to:
```
function() ImageWidget:recomputeScaleFactor(scaling)
    return self.scale_factor * scaling
end
```
(except for the cases where there is not scale_factor or it is nil or 0 :) but we could as well store the scale_factor we use the last time as a property - and may be we are already doing that.)

But it doesn't feel like documentation or defensivness: it feels like there's some clever essential computation here and that one should stop here to understand it (like I'm doing :) and well, I stopped and I'm still stopped :) The comments is still confusing. It feels like someone got lost in maths and is taking too much precautions :) which I understand :) Just sharing so you might spend more time to be sure it's needed.

Reply:
> function() ImageWidget:recomputeScaleFactor(scaling)
>     return self.scale_factor * scaling
> end

It *is* very close to that (as in, the only difference introduced is because of minute differences between w & h, the various truncation passes, and rounding errors).

> (except for the cases where there is not scale_factor or it is nil or 0 :) but we could as well store the scale_factor we use the last time as a property - and may be we are already doing that.)

It's already re-computed first in IV if it's 0.

----

I can go back to scale_factor * scaling if you think it's clearer, it won't change much in terms of actual results ;).

Reply:
I do think it's clearer - in the sense it doesn't deceive you by making you think it's doing more and different than the previous `self.scale_factor = self.scale_factor + inc` :) (while I was somehow hoping it did different :)
(ie. it's clearer having this single line in ImageViewer, that having to follow the trail into ImageWidget to somehow conclude it _only_ does this).

Reply:
Oh, it is different, since it's actually matching what rescaling the image does (i.e., `scale_factor = scale_fact * (1 + inc)` or `* (1 - inc)`, which is different than what happened previously, where we just unfiformyl added/took a chunk of the factor directly.

Reply:
Thanks. The change indeed did not jump as me, like it does now :)